src/librustc_codegen_ssa/mir/block.rs

```diff
@@ -1160,8 +1159,7 @@ impl<'a, 'tcx, Bx: BuilderMethods<'a, 'tcx>> FunctionCx<'a, 'tcx, Bx> {
         bx: &mut Bx,
         span: Span,
     ) -> Bx::Value {
-        let location = const_caller_location(bx.tcx(), bx.param_env(), span);
-
+        let location = bx.tcx().const_caller_location(span);
         bx.scalar_to_backend(
             location.val.try_to_scalar().unwrap(),
```

Comment:
Wait, is there no way to lower an entire `ty::Const` without needing this entire dance?